c++/internal/cc_capnp_library.bzl

```diff
@@ -0,0 +1,150 @@
+"Bazel rule to compile .capnp files into c++."
+
+capnp_provider = provider("Capnproto Provider", fields = {
+    "includes": "includes for this target (transitive)",
+    "inputs": "src + data for the target",
+    "src_prefix": "src_prefix of the target",
+})
+
+def _workspace_path(label, path):
+    if label.workspace_root == "":
+        return path
+    return label.workspace_root + "/" + path
+
+def _capnp_gen_impl(ctx):
+    label = ctx.label
+    src_prefix = _workspace_path(label, ctx.attr.src_prefix) if ctx.attr.src_prefix != "" else ""
```

Comment:
This stuff is now moved to `c++/internal`, so that it can easily be imported by both the public API layer (so it can be re-exported to users), and imported within the Cap'n'Proto library itself, but without asking users to import from `//src`, etc.

Also makes sure the public API surface is clean, obvious, and maintainable. Targets within the `internal` or `src` directories could therefore be made private to external parties.

Other than moving it (and the notes below), the logic for the `cc_capnp_library` rule is unmodified.